src/components/Register/CheckBox.tsx

```diff
@@ -0,0 +1,40 @@
+import styled from "@emotion/styled";
+import { InputHTMLAttributes } from "react";
+import { IcCheckMintWhite, IcCheckWhiteGrey } from "../../assets";
+
+interface CheckBoxProps extends InputHTMLAttributes<HTMLInputElement> {
+  text?: string;
+  onClick?: () => void;
+  disabled?: boolean;
+}
+
+function CheckBox({ text, onClick, disabled, checked }: CheckBoxProps): JSX.Element {
+  return (
+    <CheckBoxWrapper isChecked={checked} disabled={disabled} onClick={onClick}>
+      <CheckBoxContent>
+        {checked ? <IcCheckMintWhite /> : <IcCheckWhiteGrey />}
+        {text && <CheckBoxText>{text}</CheckBoxText>}
+      </CheckBoxContent>
+    </CheckBoxWrapper>
+  );
+}
```

Comment:
Checkbox가 공통으로 쓰이는 컴포넌트가 아니라 이 페이지에서만 쓰이는 컴포넌트 같아요.
그래서 `text`는 부모에서 prop으로 내려줄 필요가 없어보이네요!!
prop을 내려줄 필요가 없는 경우에는 최대한 내려주지 않는게 복잡한 데이터 흐름을 방지할 수 있어요!
또 `onClick`은 반드시 있어야 하는 prop이기 때문에 optional prop보다는 required prop이 더 적합해보이네요!!

disabled라는 prop은 어디에 사용되는지 궁금해요! 지금 코드를 보면 `CheckBoxWrapper`의 style prop으로 넘겨주고 있는데 막상 CheckBoxWrapper의 스타일링에는 사용되지 않는 것 같고 있네요!!

```suggestion
interface CheckBoxProps extends InputHTMLAttributes<HTMLInputElement> {
  onClick: () => void;
}

function CheckBox({ onClick }: CheckBoxProps) {
  return (
    <CheckBoxWrapper onClick={onClick}>
      <CheckBoxContent>
        {checked ? <IcCheckMintWhite /> : <IcCheckWhiteGrey />}
        <CheckBoxText>네, 모두 동의합니다.</CheckBoxText>
      </CheckBoxContent>
    </CheckBoxWrapper>
  );
}
```